### PR TITLE
[bitnami/mariadb-galera] Added SSL configuration

### PR DESCRIPTION
**Pull request by luizgn**
**Description of the change**

Added SSL certificates configuration to enable Data-in-Transit Encryption.

It uses same model of MySQL and Percona Xtradb Cluster Helm Charts.

**Benefits**

Allow this chart to be used in environments which requires Data-in-Transit Encryption.

**Possible drawbacks**

N/A. Data-in-Transit Encryption is optional and not enabled by default.

**Applicable issues**

N/A

**Additional information**

Tested with Helm 3.0.1 and Kubernetes v1.15.6.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

### CODE REVIEW

File: bitnami/mariadb-galera/README.md [hunk 1]
```diff
@@ -157,6 +157,12 @@ The following table lists the configurable parameters of the MariaDB Galera char
...
 | `metrics.serviceMonitor.selector`    | Default to kube-prometheus install (CoreOS recommended), but should be set according to Prometheus install                                                  | `{prometheus: "kube-prometheus"}`                                 |
+| `ssl.enabled`                        | Setup and use SSL for MySQL connections                                                                                                                     | `false`                                                           |
+| `ssl.secret`                         | Name of the secret containing the SSL certificates                                                                                                          | mysql-ssl-certs                                                   |
+| `ssl.certificates[0].name`           | Name of the secret containing the SSL certificates                                                                                                          | `nil`                                                             |
```
Comment (sameersbn) [inline bitnami/mariadb-galera/README.md]: You may need to update the field description

File: bitnami/mariadb-galera/templates/secrets.yaml [hunk 2]
```diff
@@ -13,3 +13,20 @@ data:
   ldap-bindpw: {{ .Values.ldap.bindpw | b64enc | quote }}
   {{- end }}
 {{- end }}
+{{- if .Values.ssl.enabled }}
+{{ if .Values.ssl.certificates }}
```
Comment (sameersbn) [inline bitnami/mariadb-galera/templates/secrets.yaml]: You can use the `and` condition here

```suggestion
{{- if and .Values.ssl.enabled .Values.ssl.certificates }}
```

File: bitnami/mariadb-galera/README.md (hunk 1 above)
Comment (luizgn) [reply bitnami/mariadb-galera/README.md]: Hi, I'm going to change MySQL to MariaDB.

Comment (sameersbn) [reply bitnami/mariadb-galera/README.md]: the `ssl.secret` and `ssl.certificates[0].name` have the exact same description.

**Comment by luizgn** [comment]
> 
> 
> Thanks for the contribution.
> From what I understand the PR creates secrets for the certificates and mounts them in the pods of the statefulset. Aren't there any mysql configuration changes required that "enable" SSL?

Hi @sameersbn, yes, to make it effective, it is required to add ssl_ca, ssl_cert and ssl_key on [mysqld] section of mariadbConfiguration using filenames defined on secret. It is also possible to add additional certificates to enable client certificate authentication.

**Comment by sameersbn** [comment]
@luizgn Instead of closing this PR, I would suggest added options to mount extraVolumes to the pods. that way a user can create the configmaps separately and mount them as  volumes in the pods. wdyt?

ref:
 - https://github.com/bitnami/charts/blob/master/bitnami/elasticsearch/templates/data-statefulset.yaml#L142-L144 
 - https://github.com/bitnami/charts/blob/master/bitnami/elasticsearch/templates/data-statefulset.yaml#L157-L159

**Comment by luizgn** [comment]
Thanks @sameersbn, excellent suggestion, it covers data-in-transit encryption, data-at-rest encryption and maybe other needs. I will add new PR to add `extraVolumes` and `extraVolumeMounts`.

**Pull request closed without merging**